common/src/View/SelectionTool.cpp

```diff
@@ -53,6 +55,48 @@ namespace TrenchBroom {
             return this;
         }
 
+        static Model::Group* enclosingGroup(Model::Node* node) {
+            if (node->parent() == nullptr) {
+                return nullptr;
+            }
+
+            auto* parentAsGroup = dynamic_cast<Model::Group*>(node->parent());
+            if (parentAsGroup != nullptr) {
+                return parentAsGroup;
+            }
+
+            return enclosingGroup(node->parent());
+        }
+
+        Model::Group* outermostClosedGroup(Model::Node* node) {
```

Comment:
I think this logic should also be placed in a visitor, or make use of the visitors if it's not general purpose enough to warrant a separate visitor. Maybe `FindGroupVisitor` could also be parameterized so that it can support this kind of query.